clients/callback_client.go

```diff
@@ -89,24 +89,42 @@ func recoverer(f func()) {
 // Sends a Transcode Status message to the Client (initially just Studio)
 // The status strings will be useful for debugging where in the workflow we got to, but everything
 // in Studio will be driven off the overall "Completion Ratio".
-func (pcc *PeriodicCallbackClient) SendTranscodeStatus(tsm TranscodeStatusMessage) {
-	if tsm.URL != "" {
-		pcc.mapLock.Lock()
-		defer pcc.mapLock.Unlock()
-
-		previousMessage, ok := pcc.requestIDToLatestMessage[tsm.RequestID]
-		previousCompletion := OverallCompletionRatio(previousMessage.Status, previousMessage.CompletionRatio)
-		newCompletion := OverallCompletionRatio(tsm.Status, tsm.CompletionRatio)
-
-		// Don't update the current message with one that represents an earlier stage
-		if !ok || tsm.IsTerminal() || newCompletion >= previousCompletion {
-			pcc.requestIDToLatestMessage[tsm.RequestID] = tsm
-		}
+func (pcc *PeriodicCallbackClient) SendTranscodeStatus(tsm TranscodeStatusMessage) error {
+	if tsm.URL == "" {
+		return nil
+	}
+	pcc.updateTranscodeStatus(tsm)
+
+	// Terminal callbacks are sent here in a sync manner
+	// Non-terminal callbacks are sent periodically, in an async manner
+	if tsm.IsTerminal() {
```

Comment:
Nice, so this also has the side-effect that we get the final callback quicker (i.e we don't have to wait for the callback interval to come around again), right?